pkg/catalog/operations/catalog.go

```diff
@@ -292,14 +291,14 @@ func (c *Catalog) splitApplicationName(applicationName string) (string, string,
 	}
 	// check if the application name has a tag
 	if strings.Contains(splited[1], ":") {
-		return "", "", nerrors.NewFailedPreconditionError("error in application name: <namespace>/<applicationName> without tag")
+		return "", "", nerrors.NewFailedPreconditionError("Error changing visibility, the visibility affects all versions of the application so not tag is allowed. Use <namespace>/<application> instead")
```

Comment:
Not sure about the name `splitApplicationName`. Seems generic, but the error makes a reference to visibility.